src/components/policy/policy_external/src/policy_helper.cc

```diff
@@ -181,24 +171,66 @@ bool policy::CheckAppPolicy::HasNewGroups(
       snapshot_->policy_table.app_policies_section.apps.find(app_policy.first);
 
   policy_table::Strings groups_new = app_policy.second.groups;
-  std::sort(groups_new.begin(), groups_new.end(), Compare);
+  std::sort(groups_new.begin(), groups_new.end(), IsDifferentStrings);
 
   policy_table::Strings groups_curr = (*it).second.groups;
-  std::sort(groups_curr.begin(), groups_curr.end(), Compare);
+  std::sort(groups_curr.begin(), groups_curr.end(), IsDifferentStrings);
 
   StringsConstItr it_groups_new = groups_new.begin();
   StringsConstItr it_groups_new_end = groups_new.end();
 
   StringsConstItr it_groups_curr = groups_curr.begin();
   StringsConstItr it_groups_curr_end = groups_curr.end();
 
+  auto IsDifferentGroupContent =
+      [this](const StringsValueType& update_group_name,
+             const StringsValueType& snapshot_group_name) -> bool {
+    if (IsDifferentStrings(update_group_name, snapshot_group_name)) {
+      return true;
+    }
+
+    const auto& func_group_from_update =
+        update_->policy_table.functional_groupings.find(update_group_name);
+    const auto& func_group_from_snapshot =
+        snapshot_->policy_table.functional_groupings.find(snapshot_group_name);
+
+    const auto& update_fg_rpcs = func_group_from_update->second.rpcs;
+    const auto& snapshot_fg_rpcs = func_group_from_snapshot->second.rpcs;
+
+    if (update_fg_rpcs.is_null() || snapshot_fg_rpcs.is_null()) {
+      return !(update_fg_rpcs.is_null() && snapshot_fg_rpcs.is_null());
+    }
+
+    if (update_fg_rpcs.size() != snapshot_fg_rpcs.size()) {
```

Comment:
@jacobkeeler why should change this logic?
In case if `update` group contains more RPCs then `snapshot` it means that some additional  RPC's are allowed for the group and `OnPermissionCHanges` notification should be sent to the mobile.

Reply:
Again, look at the usage of [std::set_difference](https://en.cppreference.com/w/cpp/algorithm/set_difference), this function isn't supposed to just determine if two values are equal.

Reply:
Please see updated implementation

Reply:
@jacobkeeler group content updates are moved to a separate method in 01ffc5e7885714b802eac2778c7c9019f0ab2e8a